### DISCUSSION TITLE
Is there a built-in submodule for penalized logp?

### DISCUSSION THREAD

**Discussion by noncomputable** [rdkit/rdkit#3918] [Development]
It would be convenient if there was a built-in module for computing either the penalized log p directly, or for computing the synthetic accessibility score and number of long cycles that define the penalized log p. But as far as I can tell none of these are conveniently bundled rdkit. Why is that? Or did I miss it when looking through the docs?

**Comment by greglandrum** [comment]
The Ertl SA Score is available in the Contrib package.
I'm not sure what the other components of "penalized logp" are; do you have a reference which actually defines how to calculate that?

**Comment by noncomputable** [comment]
Thanks, I see it! I'm following the definition here https://www.nature.com/articles/s41598-019-47148-x.pdf (page 5, first par of "single property optimization").

**Reply by greglandrum** [reply]
I don't see a useable definition in there. They say "Penalized logP is the logP minus the synthetic accessibility (SA)
score and the number of long cycles." but they don't say which SA score they use or what a "long cycle" is. There's a reference, but I didn't see an actual definition in that either.

**Reply by noncomputable** [reply]
The reference defines long cycles right below where they first mention it under section 3.2 here: https://arxiv.org/pdf/1802.04364.pdf
